### PR TITLE
Fix deprecation warnings, remove v0.3 support

### PR DESCRIPTION
**Pull request by ranjanan**
Add REQUIRE file to tests for StatsBase, because `hist` is deprecated, and `sub` is now `view`

### CODE REVIEW

File: src/SortingAlgorithms.jl [hunk 1]
```diff
@@ -1,6 +1,7 @@
 module SortingAlgorithms
 
 using Compat
+import Compat.view
```
Comment (tkelman) [inline src/SortingAlgorithms.jl]: put a minimum version on Compat in REQUIRE. you shouldn't need me to tell you this again.

**Comment by ranjanan** [comment]
The reason this is failing is because StatsBase does not support v0.3. I used StatsBase because the `hist` function is deprecated in Base. Would it be okay to drop v0.3 support for this package?

**Comment by tkelman** [comment]
I think dropping 0.3 support here is fine.

**Comment by ranjanan** [comment]
@kmsquire @tkelman @ViralBShah Could one of you guys tag a new version? Thanks.

**Comment by tkelman** [comment]
(will have to be a new minor version via `Pkg.tag("SortingAlgorithms", :minor)` since 0.3 support was dropped)

**Pull request merged**